### PR TITLE
Add a new layer ElementwiseLambdaLayer

### PR DESCRIPTION
**Pull request by One-sixth**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

File: tensorlayer/layers/merge.py [hunk 1]
```diff
@@ -150,3 +151,56 @@ def __init__(
...
+    >>> mean = tl.layers.DenseLayer(net, n_units=200, name='mean')
+    >>> std = tl.layers.DenseLayer(net, n_units=200, name='std')
+    >>> z = tl.layers.ElementwiseLambdaLayer([noise, mean, std], fn=func, name='z')
+    """
+
+    def __init__(
+            self,
+            layers,
+            fn,
+            fn_args=None,
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/merge.py]: please add parameter
-  act=None

File: tensorlayer/layers/merge.py [hunk 2]
```diff
@@ -150,3 +151,56 @@ def __init__(
...
+    ):
+
+        super(ElementwiseLambdaLayer, self).__init__(prev_layer=layers, name=name)
+        logging.info("ElementwiseLambdaLayer %s" % self.name)
+
+        if fn_args is None:
+            fn_args = {}
+
+        self.inputs = [layer.outputs for layer in layers]
+        with tf.variable_scope(name) as vs:
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/merge.py]: Add a linebreak before the new scope

File: tensorlayer/layers/merge.py [hunk 3]
```diff
@@ -150,3 +151,56 @@ def __init__(
...
+            self.outputs = fn(*self.inputs, **fn_args)
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/merge.py]:
```python
self.outputs = fn(*self.inputs, **fn_args)
if act:
    self.outputs = act(self.outputs)
```

**Comment by One-sixth** [comment]
I have finished modifying it.

File: CHANGELOG.md [hunk 4]
```diff
@@ -76,6 +76,8 @@ To release a new version, please update the changelog as followed:
 - CI Tool:
   - Danger CI has been added to enforce the update of the changelog (by @lgarithm and @DEKHTIARJonathan in #563)
   - https://github.com/apps/stale/ added to clean stale issues (by @DEKHTIARJonathan in #573)
+- API:
```
Comment (DEKHTIARJonathan) [inline CHANGELOG.md]: Change for **Layer**

File: CHANGELOG.md [hunk 5]
```diff
@@ -76,6 +76,8 @@ To release a new version, please update the changelog as followed:
...
+  - Add new layer ElementwiseLambdaLayer (by @One-sixth in #576)
```
Comment (DEKHTIARJonathan) [inline CHANGELOG.md]: Incorrect PR number => #579 

Please avoid using 'add of blablabla'. We already know you added something thanks to the Changelog Section.

**Make smthg like this:**
- `ElementwiseLambdaLayer` to perform ????????????????? (by @One-sixth in #579)

File: tensorlayer/layers/merge.py [hunk 6]
```diff
@@ -150,3 +151,60 @@ def __init__(
         # # self.all_drop = list_remove_repeat(self.all_drop)
 
         self.all_layers.append(self.outputs)
+
+
+class ElementwiseLambdaLayer(Layer):
+    """A layer uses a custom function to join multiple layer inputs.
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/merge.py]: English Syntax: A layer that use a custom function to combine multiple :class:`Layer` inputs.

File: tensorlayer/layers/merge.py [hunk 7]
```diff
@@ -150,3 +151,60 @@ def __init__(
...
+
+    Parameters
+    ----------
+    layers : list of :class:`Layer`
+        The list of layers to combine.
+    fn : function
+        The function that applies to the outputs of previous layer.
+    fn_args : dictionary or None
+        The arguments for the function (option).
+    name : str
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/merge.py]: **Missing:**
```python
act : activation function
        The activation function of this layer.
```

**Comment by DEKHTIARJonathan** [comment]
@One-sixth I was first focusing on the "cosmetic" aspects.

Now I see two major issues before merging:
1. The layer you implemented is missing some unittest to make sure everything works as planned: [test_layers_merge.py](https://github.com/tensorlayer/tensorlayer/blob/master/tests/test_layers_merge.py)
2. Layers in Tensorlayers use **only** TL layers or need to be wrapped into `InputLayer`, your implementation do not follow this principal: https://github.com/tensorlayer/tensorlayer/blob/ea8cc396495e6c0cbb368e24febebadab9f501d5/tensorlayer/layers/merge.py#L63

File: CHANGELOG.md [hunk 8]
```diff
@@ -76,6 +76,8 @@ To release a new version, please update the changelog as followed:
 - CI Tool:
   - Danger CI has been added to enforce the update of the changelog (by @lgarithm and @DEKHTIARJonathan in #563)
   - https://github.com/apps/stale/ added to clean stale issues (by @DEKHTIARJonathan in #573)
+- Layer:
+  - ElementwiseLambdaLayer added to use custom function to connect multiple layer inputs. (by @One-sixth in #579)
```
Comment (DEKHTIARJonathan) [inline CHANGELOG.md]: remove "." before  "(by @One-sixth in #579)"

**Comment by DEKHTIARJonathan** [comment]
Error with YAPF, run the following:
yapf -i --style=setup.cfg tests/test_layers_merge.py

**Comment by DEKHTIARJonathan** [comment]
@One-sixth congratulation on this PR! I merge immediately :D

**Pull request merged**